examples/scripts/delete_metric_descriptors/delete_metric_descriptors.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+	"unicode"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+/*
+This script deletes metric descriptors from the given projects (-projects flag),
+matching the given metric type (descriptor name) regex expression (-metric_type_regex flag).
+
+Metrics to delete will be first printed and then awaiting interactive confirmation,
+before the actual removal. Dry run option also exists.
+
+WARNING: All underlying time series behind each descriptor (potentially years
+of data) will be irreversibly removed once confirmed.
+
+Example run:
+
+1. Setup Application Default Credentials (ADC) (https://cloud.google.com/docs/authentication/provide-credentials-adc)
+if you haven't yet:
+	1a. Make sure the account behind the ADC for chosen projects has Monitoring Editor or Monitoring Admin permissions: https://cloud.google.com/monitoring/access-control#monitoring-perms
+  1b. Acquire Application Default Credentials in your environment using gcloud:
+
+gcloud auth application-default login
+
+2. Run Go script (from the same directory as the script):
+
+go run delete_metric_descriptors.go -projects projects/<your-project> -metric_type_regex "<your matching expression>"
+
+See go run delete_metric_descriptors.go -help for all options.
+*/
+
+var (
+	cloudMonitoringEndpoint = flag.String("address", "monitoring.googleapis.com:443", "address of monitoring API")
+
+	projectNames    = flag.String("projects", "", "required: comma-separated project IDs of the projects on which to execute the requests. Name format is as defined in https://cloud.google.com/monitoring/api/ref_v3/rpc/google.monitoring.v3#listmetricdescriptorsrequesttarget, e.g. projects/test-project,projects/test-project2")
+	metricTypeRegex = flag.String("metric_type_regex", "", "required: RE2 regex expression matching metric.type (anchored), so metric descriptor names to delete. Guarded with the interactive 'y' confirmation. See --dry_run to only print those")
+	dryRun          = flag.Bool("dry_run", false, "whether to dry run or not")
+
+	serviceAccountEnvVar = flag.String("sa-envvar", "", "optional environment variable containing Google Service Account JSON, without it application-default flow will be used.")
+)
+
+func deleteDescriptors(endpoint string, projects []string, re2 *regexp.Regexp, saEnvVar string, dryRun bool) error {
```

Comment:
nit: why is it called `re2`? Where is `re1`?

Reply:
It's for https://github.com/google/re2/wiki/Syntax but can change 🙈

Reply:
Fixed!